Add tf.regex_match for regex match support

yongtang opened the pull request:
This fix tries to address the issue raised in #18264.

Currently tf.regex_replace has already been supported though there was no regex match support.
This fix adds the tf.regex_match support in a similiar pattern as tf.regex_replace.

This fix fixes #18264.

Note: this PR is a preliminary implementation which support full match of the regex. Additional variations could be added or in follow up PRs if needed.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Add tf.regex_match for regex match support

This fix tries to address the issue raised in 18264.
Currently tf.regex_replace has already been supported
though there was no regex match support.
This fix adds the tf.regex_match support in a similiar
pattern as tf.regex_replace.

This fix fixes 18264.

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Update BUILD file for the tf.regex_match kernel

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Register RegexMatch ops

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Add test cases for tf.regex_match

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Update api_defs

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Update API golden

update with:
```
bazel-bin/tensorflow/tools/api/tests/api_compatibility_test
           --update_goldens True
```

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

rmlarsen commented on tensorflow/core/api_def/base_api/api_def_RegexMatch.pbtxt:
```diff
@@ -0,0 +1,22 @@
+The same tensor shape as `input`.
+END
+  }
+  summary: "Check if the input matches the regex pattern."
```
Can you expand the documentation to describe the shapes and nature of the output. As I read it, the op takes a string tensor of any shape, and a scalar string tensor containing a single regex pattern. It returns a boolean tensor that of the same shape as the input, with values True for those inputs that match the pattern. Perhaps say that this expect regular expressions in the default format used by RE2 and give a link to the description of this.

yongtang replied:
Thanks @rmlarsen. The doc has been updated.

asimshankar reviewed (commented):
(From API review)
Thanks for the PR! 

- The op should be `RegexFullMatch` 
- The Python function should be `regex_full_match` (The "full" more clearly calls out that this is different from Python's `re.match`)
- Let's put this in the `tf.strings` namespace

yongtang committed:
Expose regex_full_match in tf.strings namespace

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Update golden API

```
bazel-bin/tensorflow/tools/api/tests/api_compatibility_test
           --update_goldens True
```

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang commented:
Thanks @asimshankar @rmlarsen for the review. The PR has been updated with op exposed as `tf.strings.regex_full_match`. Please take a look and let me know if there are any issues.

rmlarsen commented:
@asimshankar handing to you for API approval.

asimshankar commented:
Good for API addition. Thanks!

rmlarsen commented:
Thanks!
